iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/pipe/PipeTaskInfo.java

```diff
@@ -283,6 +341,19 @@ public TSStatus createPipe(CreatePipePlanV2 plan) {
     }
   }
 
+  public TSStatus alterPipe(AlterPipePlanV2 plan) {
+    acquireWriteLock();
+    try {
+      pipeMetaKeeper.removePipeMeta(plan.getPipeStaticMeta().getPipeName());
+      pipeMetaKeeper.addPipeMeta(
+          plan.getPipeStaticMeta().getPipeName(),
+          new PipeMeta(plan.getPipeStaticMeta(), plan.getPipeRuntimeMeta()));
```

Comment:
I strongly suggest that updating progressIndex also here... What about DNs are crushed? Then they restart and fetch only a MinimumProgressIndex at configNode...